#162170677 Get all article with pagination support

Johnsonojo opened the pull request:
#### What does this PR do?
It gets all article and has pagination implemented


#### Description of Task to be completed?
- add actions and reducers for getting all articles
- create all articles component
- add pagination to navigate between pages


#### How should this be manually tested?
- run `npm run start:dev` and navigated to `hosturl/articles`
- click the button to get more articles


#### Any background context you want to provide?
N/A


#### What are the relevant pivotal tracker stories? (If applicable)
[#162170677](https://www.pivotaltracker.com/story/show/162170677)


#### Screenshots (If applicable)
<img width="647" alt="pagination 1" src="https://user-images.githubusercontent.com/32887459/51958832-3f558c00-2452-11e9-9dc2-034d11dc3b7b.png">

<img width="343" alt="pagination 2" src="https://user-images.githubusercontent.com/32887459/51958833-3f558c00-2452-11e9-8e92-ed26f80d4d1f.png">
<img width="684" alt="pagination 3" src="https://user-images.githubusercontent.com/32887459/51958834-3fee2280-2452-11e9-90d7-9c561d0845ee.png">



#### Questions:
N/A

frostyblok commented on src/components/articles/AllArticles.jsx:
```diff
@@ -0,0 +1,106 @@
+          <div className="card">
+            <div className="card-image">{article.imgUrl}</div>
+            <div className="card-image">
+              {/* <img src="../../assets/images/typewriter.jpg" alt="FakeImage" /> */}
```
Remove commented out code.

frostyblok commented on src/components/articles/AllArticles.jsx:
```diff
@@ -0,0 +1,106 @@
+    this.setState({ done: true });
+  };
+
+  renderArticles() {
```
I saw that you were calling this on line 78, have you thought about making the elements in this method a component on its own? I would like to get your response on this

Johnsonojo replied:
Hi @frostyblok, It has been refactored to a reusable component

frostyblok commented on src/reducers/articles.js:
```diff
@@ -0,0 +1,20 @@
+    case GET_ALL_ARTICLES_FAILURE_MSG:
+      return { ...state, message: action.payload };
+    default:
+      return state;
```
Thank you for now mutating the state here.

Johnsonojo replied:
@frostyblok, I did not mutate state here 😆

frostyblok reviewed (commented):
Hi Johnson, you have really done a great job! I have left you some comments, kindly go through them

omokaro-faith commented:
@Johnsonojo First things first, Please fix hounds violation.

akhilome commented on src/components/article/AllArticles.jsx:
```diff
@@ -0,0 +1,118 @@
+            </h1>
+          </div>
+        </div>
+        {/* <div className="container"> */}
```
Please take out this comment if this line is no longer being utilized

Johnsonojo replied:
Thanks for the observation. I have removed it.

akhilome commented on src/components/article/AllArticles.jsx:
```diff
@@ -0,0 +1,118 @@
+            {'See more >>>'}
+          </button>
+        </div>
+        {/* </div> */}
```
Here too

Johnsonojo replied:
I have fixed it. Thanks

akhilome commented on src/components/article/AllArticles.jsx:
```diff
@@ -0,0 +1,118 @@
+    const { done, defaultImg, defaultText } = this.state;
+    if (done) {
+      return allArticles.allArticles.map(article => (
+        <div className="col s12 m6 l6" key={article.id}>
```
This should be abstracted out into its own presentational component like, say, `ArticleCard`  since it's not only on this view that we have this form of article card.

Creating a presentation component with receives props for the kind of content to show on the card, would make it easier for every other view (like related articles section on the article page, homepage, profile page) to reuse it.

Johnsonojo replied:
Working on this already.

Johnsonojo replied:
@akhilome I have already refactored it to be a reusable component.

akhilome commented on src/components/article/AllArticles.jsx:
```diff
@@ -0,0 +1,118 @@
+            </div>
+
+            <div className="card-content">
+              <span className="card-title">{article.title}</span>
```
Quick observation:

while perusing your awesome work on this feature and testing out the app, I noticed that one can not click on an article card (or even the title) to read the article.

Here would be a good spot to `Link to` to the article itself don't you think?

Johnsonojo replied:
Nice observation. It has been fixed

akhilome reviewed (commented):
@Johnsonojo 👋 what's good?

Good job delivering on this task 🎉 

I left a few comments below, please go over them.

Feel free to reach out if you need more clarification on anything 💪

omokaro-faith commented:
Sorry you have conflicts again.

akhilome reviewed (approved):
Save for hound violations, this PR is finally good to go.

Awesome job working and delivering on the errors I pointed out to you in person. 💪

omokaro-faith commented on src/components/article/AllArticles.jsx:
```diff
@@ -0,0 +1,107 @@
+              <p>
+                {
+                  '"The more that you read, the more things you will know. The more you learn, the more places you will go"'
+                }
```
@Johnsonojo No need to add `{
                  '"The more that you read, the more things you will know. The more you learn, the more places you will go"'
                }`

You can add it as plain text or you can use template literal.

Johnsonojo replied:
Alright. I will refactor it

Johnsonojo replied:
Hi @omokaro-faith, I have refactored it.

omokaro-faith commented on src/components/presentation/ArticleCard.jsx:
```diff
@@ -0,0 +1,41 @@
+import PropTypes from 'prop-types';
+
+const ArticleCard = ({ article }) => {
+  const defaultImg = 'https://tmsnrt.rs/2FNDBT2';
```
I wouldn't want you exposing this. What about adding it to the .env file and calling it here as maybe process.env.REACT_DEFAULT_IMG?

Johnsonojo replied:
Thanks for the observation. I will refactor it.

Johnsonojo committed:
[Feature #162170677] Get all article with pagination support

- add actions and reducers for all articles
- create all articles component

[Finishes #162170677]
